File: src/main/java/com/github/jarol/azure/search/spark/sql/connector/write/ConstantActionSupplier.java
package com.github.jarol.azure.search.spark.sql.connector.write;

import com.azure.search.documents.models.IndexActionType;
import org.apache.spark.sql.catalyst.InternalRow;
import org.jetbrains.annotations.Contract;

/**
 * Constant supplier (used when no index action column has been specified)
 */

public class ConstantActionSupplier
        implements IndexActionSupplier {

    private final IndexActionType defaultValue;

    /**
     * Create a new instance with a default value
     * @param defaultValue value
     */

    @Contract(pure = true)
    public ConstantActionSupplier(
            IndexActionType defaultValue
    ) {
        this.defaultValue = defaultValue;
    }

    @Override
    public IndexActionType get(InternalRow row) {
        return defaultValue;
    }
}

File: write/src/main/scala/com/github/jarol/azure/search/spark/sql/connector/write/ConstantActionSupplier.scala
package com.github.jarol.azure.search.spark.sql.connector.write

import com.azure.search.documents.models.IndexActionType
import org.apache.spark.sql.catalyst.InternalRow

/**
 * Supplier that provides a constant value for all [[InternalRow]](s)
 * @param default default value
 */

case class ConstantActionSupplier(private val default: IndexActionType)
  extends IndexActionSupplier {

  override def get(row: InternalRow): IndexActionType = default
}
